### ISSUE TITLE
Question: Could NOT find GDAL (missing: GDAL_INCLUDE_DIR) under CentOS

### ISSUE THREAD

**Issue by twilkinson** [dtarb/TauDEM#172]
I'm trying to get TauDEM running under CentOS 7 (following the commands shown in the github README) but when I get to the command

`cmake ..` 

I get the error

`Could NOT find GDAL (missing: GDAL_INCLUDE_DIR)`

I created an environment variable with

`GDAL_INCLUDE_DIR=/path/to/gdal/2.1.3/include`
`EXPORT GDAL_INCLUDE_DIR`

but when I run `cmake `again, I still get the error.

I'm somewhat new to Linux and I expect the fix is something simple, but can someone give me a hint as to what I'm missing?

Thanks,
Terry

**Comment by mach0** [comment]
Hi!
It's not enough to set "any" Path - the "/path/to/gdal/2.1.3/include" looks very randomly set.
Not much experience with CentOS here but on debian you have to install the package "libgdal-dev" and the PATH to the include files is "/usr/include/gdal/" - you might have to find the correct package for CentOS. 
Hope this will help

**Comment by twilkinson** [comment]
@mach0, thanks for your help.  I should have been more specific in my initial post.  The path value in I specified was meant to be just an indicator for the actual path to the include files.  On our system, `gdal` is loaded using the _lmod module system_, so the actual path is quite long.  The actual commands I used are:

`GDAL_INCLUDE_DIR=/cvmfs/soft.computecanada.ca/easybuild/software/2017/avx2/Compiler/intel2016.4/gdal/2.1.3/include`
`EXPORT GDAL_INCLUDE_DIR`

It seems to me from the error message that it can't find an entry for `GDAL_INCLUDE_DIR`, as opposed to finding the entry but not finding the actual files.

**Comment by mach0** [comment]
ok I see.. just out of interest (because there is a "Compiler" directory: Is that the path to the sources or is GDAL installed there?
Another hint: Have you ever tried ccmake to see which paths are automatically found and where you have to "help" a little bit ;) - I think your path will never be found automatically ;)

**Comment by twilkinson** [comment]
I think GDAL is installed there (not the sources, but it does have an `include` directory with the `*.h` files).

![image](https://user-images.githubusercontent.com/1958073/40274869-d00f61ba-5baf-11e8-8ab9-e538b05a5180.png)

I poked around a bit more and it looks like the `cmake` command generated a file called `CMakeCache.txt`.  Looking through that file, I found a line that looked like it was trying to define `GDAL_INCLUDE_DIR`.  I edited it to point to the directory with the `*.h` files as follows:

`GDAL_INCLUDE_DIR:PATH=/cvmfs/soft.computecanada.ca/easybuild/software/2017/avx2/Compiler/intel2016.4/gdal/2.1.3/include`

and then retried the `cmake ..` command -  and it looks like it worked!!

I think your comment about "path will never be found automatically" was what got me thinking about that generated file and set me on the right course.

Thanks very much for your help!  Have a good weekend!
